File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k)
    {
        int res = INT_MAX;
        sort(nums.begin(), nums.end());

        for (int i = 0; i < nums.size(); i++)
        {
            if (i + k - 1 < nums.size())
                res = min (res, nums[i + k - 1] - nums[i]);
        }
        return res;
        
    }
};

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        res = nums[-1]
        for i, num in enumerate(nums):
            if i + k - 1 < len(nums):
                res = min (res, nums[i + k - 1] - nums[i])
            i += 1
        return res
            
        